### PR TITLE
Documentation strings for commands.rs - spawn, despawn, with, with_bundle, insert, insert_one

### PR DESCRIPTION
**Pull request by eXodiquas**
I wrote some documentation for the methods on Commands that I find myself using often. And as I started learning bevy those are methods I would have liked to just have a small example directly in the docs, so I made them. Let me know if something is not correct or if something could be written better. Maybe I wrote it too beginner friendly and lost some of the use cases of the methods that I have not encountered yet.

If someone tells me what the undocumented methods do, I'm happy to add those doc strings aswell. :)

### CODE REVIEW

File: crates/bevy_ecs/src/system/commands.rs [hunk 1]
```diff
@@ -223,6 +223,13 @@ pub struct Commands {
 }
 
 impl Commands {
+    /// Creates a new entity with a given component in the current [World].
```
Comment (cart) [inline crates/bevy_ecs/src/system/commands.rs:226]: This is incorrect. This doesn't create an entity with a given component, but rather with a given DynamicBundle (which is a "set" of components).

It is worth showing two examples here: a spawn example using tuples:

```rust
commands.spawn((MyComponentA, MyComponentB))
```

And an example using a derived Bundle

```rust
#[derive(Bundle)]
struct MyBundle {
  a: MyComponentA,
  b: MyComponentB,
}

commands.spawn(MyBundle {
  a: MyComponentA,
  b: MyComponentB,
})
```

Comment (cart) [reply crates/bevy_ecs/src/system/commands.rs:226]: And lets call out that the two examples are equivalent.

File: crates/bevy_ecs/src/system/commands.rs [hunk 2]
```diff
@@ -248,10 +255,24 @@ impl Commands {
...
+    /// # Example
+    /// ```
+    /// commands
+    ///     .despawn(EntitiyToDespawn);
+    /// ```
     pub fn despawn(&mut self, entity: Entity) -> &mut Self {
         self.write_world(Despawn { entity })
     }
 
+    /// Describes an entity with an additional component.
```
Comment (cart) [inline crates/bevy_ecs/src/system/commands.rs:268]: Lets rephrase this to "Adds the given component to the current entity"

File: crates/bevy_ecs/src/system/commands.rs [hunk 3]
```diff
@@ -271,6 +309,13 @@ impl Commands {
         self
     }
 
+    /// Adds components to a already existing entity.
```
Comment (cart) [inline crates/bevy_ecs/src/system/commands.rs:312]: "Adds a component bundle to an already existing entity"

File: crates/bevy_ecs/src/system/commands.rs [hunk 4]
```diff
@@ -279,6 +324,18 @@ impl Commands {
         self.write_world(Insert { entity, components })
     }
 
+    /// Adds a single component to a already existing entity.
```
Comment (cart) [inline crates/bevy_ecs/src/system/commands.rs:327]: nit: "an already"

**Review by cart — CHANGES_REQUESTED** [review]
Thanks for docs! This will definitely help new people learn how to use commands (which is a common issue)

File: crates/bevy_ecs/src/system/commands.rs [hunk 5]
```diff
@@ -223,6 +223,29 @@ pub struct Commands {
 }
 
 impl Commands {
+    /// Creates a new entity with a given [DynamicBundle] (which is a 'set' of components) in the current [World].
+    /// 
+    /// # Example
+    /// ```
+    /// // An example with only one component
+    /// commands.spawn(MyComponent);
```
Comment (cart) [inline crates/bevy_ecs/src/system/commands.rs:231]: you can't spawn single components like this. it must be a tuple. Can we replace it with:

```rust
commands.spawn((MyComponent,));
```
Lets fix this in the other examples too.

**Review by cart — CHANGES_REQUESTED** [review]

File: crates/bevy_ecs/src/system/commands.rs (hunk 5 above)
Comment (eXodiquas) [reply crates/bevy_ecs/src/system/commands.rs]: Ah, damn. I was deceived by 'SpriteComponents' but it is a Bundle. I always wondered why there was a trailing 's'. Now I know. Should be fine now. (hopefully)

Comment (cart) [reply crates/bevy_ecs/src/system/commands.rs]: Yeah you arent the first person. I'm thinking about replacing the "_Components" suffix with "_Bundle" for clarity.

**Review by cart — APPROVED** [review]
Awesome lets merge this!

**Comment by cart** [comment]
Ah first we need to resolve the formatting conflicts (you can see the suggested fixes in the build logs)

Looks like you just need to remove the extra newlines from the docs.

**Comment by eXodiquas** [comment]
I believe it is not in my power to fix the remaining 5 build actions. Looks like the nightly version of rust is not installed successfully.

**Comment by CleanCut** [comment]
I bet CI would succeed now. It's been a month of nightlies. 😄

**Comment by cart** [comment]
Its interesting that `norun` still caused doctest failures. I'm thinking we should probably just add unit structs and hide them from the docs using this approach: https://doc.rust-lang.org/rustdoc/documentation-tests.html#hiding-portions-of-the-example

That way we still get protection against breakage.

**Comment by memoryruins** [comment]
The examples will need to be wrapped in some boilerplate since the code is still fully built with no_run and doc tests are treated as external to the current crate, e.g.
```rust
/// Despawns only the specified entity, ignoring any other consideration.
/// # Example
/// ```no_run
/// # use bevy_ecs::{Commands, Entity};
/// # struct EntitiyToDespawn;
/// # fn system(mut commands: Commands) {
/// # let entity_to_despawn = Entity::new(0);
/// commands.despawn(entity_to_despawn);
/// # }
/// ```
pub fn despawn(&mut self, entity: Entity) -> &mut Self {
```
Another option is to use `ignore`, but that has no protection from breakage and shows a warning on each example in the docs, so it's usually not preferable.

**Comment by cart** [comment]
Cool then no-run w/ the additional boilerplate hidden seems like the right call

**Comment by eXodiquas** [comment]
Aye, I'm going to build some boilerplate then. I hope I get some spare time on my sunday.

**Comment by DJMcNab** [comment]
Is there a status update on thie @eXodiquas?

**Comment by cart** [comment]
Closed in favor of #976
